scripts/codemods/typed_flags.py

```diff
@@ -0,0 +1,182 @@
+                    """
+                    if TYPE_CHECKING:
+                        @_generated
+                        def __init__(self):
+                            ...
+                    """
+                )
+                if_block = cst.parse_statement(code)  # type: ignore
+                assert if_block  # noqa: S101
+                # now we need to add this if_block into the CST of node
+                # find the first function definition and insert it before there
+                body = list(node.body.body)
+                for pos, b in enumerate(body):
+                    if m.matches(b, m.FunctionDef()):
+                        body.insert(pos, if_block)
+                        node = node.with_deep_changes(node.body, body=body)
+                        break
+                else:
+                    # so there wasn't any function, not sure why
+                    # doesn't really matter, we just insert at the end
+                    body.append(if_block)
+                    node = node.with_deep_changes(node.body, body=body)
+
+            # find the init
+            for b in if_block.body.body:
+                if m.matches(b, m.FunctionDef(m.Name("__init__"))) and isinstance(
+                    b, cst.FunctionDef
+                ):
+                    init = b
+                    break
+            else:
+                raise RuntimeError
+            # add to the init the generated decorator
+            old_init = init
+            init = old_init.with_changes(decorators=[cst.Decorator(cst.Name("_generated"))])
+            node = node.deep_replace(old_init, init)  # type: ignore
+            node = node.with_deep_changes(init.params, kwonly_params=kwonly_params, star_kwarg=None)
+
+        else:
+            # the init exists, so we need to make overloads
+            # here's how:
+            # first, we use the init we've already found
+            # now, we need to insert two overloaded functions before it
+            # this can be done by getting an init `with_changes` and remove the body
+            # then we add the _generated decorator
+            # then we remove all parameters except for self, and save that as our first overload
+            # next, we need to add all of the flag values and that is our second overload
+            # then we put them together and insert them before the existing overload
+            # todo: this is very complicated, and maybe this visitor should be split into another visitor just to do this logic
+            assert init  # noqa: S101
+            no_body_init = init.with_changes(
+                body=cst.IndentedBlock([cst.SimpleStatementLine([cst.Expr(cst.Ellipsis())])]),
+                decorators=[
+                    cst.Decorator(cst.Name("overload")),
+                    cst.Decorator(cst.Name("_generated")),
+                ],
+            )
+
+            # todo: remove all parameters except the first
```

Comment:
no need, the point of this is just so that type-checkers don't complain about only having one overload; this overload isn't meant to be used

Reply:
https://github.com/DisnakeDev/disnake/pull/778/commits/d91247e3808950b841a9b0f97ca019408a6c48e0